core/src/commonMain/kotlin/io/github/aag/core/data/network/repos/CourseRepositoryImpl.kt

```diff
@@ -0,0 +1,43 @@
+package io.github.aag.core.data.network.repos
+
+import io.github.aag.core.data.network.models.toCourse
+import io.github.aag.core.data.network.sources.CoursesRemoteDataSource
+import io.github.aag.core.domain.OperationResult
+import io.github.aag.core.domain.models.Course
+import io.github.aag.core.domain.repositories.CourseRepository
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+class CourseRepositoryImpl(
+    private val remoteDataSource: CoursesRemoteDataSource
+) : CourseRepository {
+
+    private val _allCourses =
+        MutableStateFlow<OperationResult<List<Course>>>(OperationResult.Loading)
+
+    override val allCourses: StateFlow<OperationResult<List<Course>>> =
+        _allCourses.asStateFlow()
+
+    override suspend fun loadAllCourses(): StateFlow<OperationResult<List<Course>>> {
+        try {
```

Comment:
```
_allCourses.value = runCatching {
   ...
}
```
will be better here

Reply:
`runCatching`  has `kotlin.Result` as a return type, I couldn't find a way to extract `OperationResult` from it.

Reply:
Choose one of these blocks

```
_allCourses.value = runCatching {
    OperationResult.Success(remoteDataSource.getAllCourses().map { entity -> entity.toCourse() })
}.getOrElse {
    OperationResult.Error(it)
}
```
or
```
_allCourses.value = try {
    OperationResult.Success(remoteDataSource.getAllCourses().map { entity -> entity.toCourse() })
} catch (t: Throwable) {
    OperationResult.Error(t)
}
```

Reply:
Yeah, indeed, this code might've been improved, thanks, done